packages/sdk-v2/src/sdk/mpbridge/hooks.ts

```diff
@@ -0,0 +1,716 @@
+  handleError,
+  getChainName,
+  getTargetChainId,
+  getSourceChainId,
+  calculateBridgeFees
+} from "./constants";
+import { CONTRACT_TO_ABI } from "../base/sdk";
+import { SupportedChains } from "../constants";
+
+/**
+ * Maps SDK environment names to mpb.json deployment names for each chain
+ */
+const getDeploymentName = (baseEnv: string, chainId: number): string => {
+  // For Fuse chain (122)
+  if (chainId === SupportedChains.FUSE) {
+    // production, staging, or fuse all use "fuse" deployment
+    return "fuse";
+  }
+
+  // For Celo chain (42220)
+  if (chainId === SupportedChains.CELO) {
+    // production and staging use "celo", development uses "alfajores" testnet
+    if (baseEnv === "production" || baseEnv === "staging") {
+      return "celo";
+    }
+    return "celo"; // Default to celo mainnet
+  }
+
+  // For Ethereum mainnet (1)
+  if (chainId === SupportedChains.MAINNET) {
+    return "mainnet";
+  }
+
+  // Fallback
+  console.warn(`Unknown chain ID ${chainId}, defaulting to mainnet deployment`);
+  return "mainnet";
+};
+
+/**
+ * Hook to get MPB Bridge contract for a specific chain
+ * Uses centralized ABI but MPB-specific addresses from @gooddollar/bridge-contracts mpb.json
+ * Dynamically selects the correct contract address based on the current environment
+ */
+const useGetMPBContract = (chainId?: number) => {
+  const { library } = useEthers();
+  const { baseEnv } = useGetEnvChainId();
+  const mpbABI = CONTRACT_TO_ABI["MPBBridge"]?.abi || [];
+  const targetChainId = chainId || BRIDGE_CONSTANTS.DEFAULT_CHAIN_ID;
+
+  return useMemo(() => {
+    if (!library) return null;
+
+    // Get the deployment name for this chain and environment
+    const deploymentName = getDeploymentName(baseEnv, targetChainId);
+
+    // Get the contract address from mpb.json
+    const contractAddress = getMPBContractAddress(targetChainId, deploymentName);
+
+    if (!contractAddress) {
+      console.error(
+        `No MPB bridge contract found for chain ${targetChainId} in environment ${baseEnv} (deployment: ${deploymentName})`
+      );
+      return null;
+    }
+
+    console.log(`Using MPB bridge contract for chain ${targetChainId}, env ${baseEnv}: ${contractAddress}`);
+    return new ethers.Contract(contractAddress, mpbABI, library);
+  }, [library, mpbABI, targetChainId, baseEnv]);
+};
+
+const getBridgeRequestTopic = (): string => {
+  // Calculate the keccak256 hash of the event signature
+  return ethers.utils.id("BridgeRequest(address,address,uint256,uint256,uint256,uint8,uint256)");
+};
+
+// Types for better readability
+interface BridgeLimits {
+  dailyLimit: ethers.BigNumber;
+  txLimit: ethers.BigNumber;
+  accountDailyLimit: ethers.BigNumber;
+  minAmount: ethers.BigNumber;
+  onlyWhitelisted: boolean;
+}
+
+interface ValidationResult {
+  isValid: boolean;
+  reason: string;
+  errorMessage?: string;
+}
+
+export const useMPBBridgeLimits = (amount: string, chainId?: number): ValidationResult => {
```

Comment:
I see two similar hooks. this one and useMPBBRidgeData

this one actually verify's canClaim (did someone use up their limits for the day)
it does not do an allowance check, which have to be done seperately against the token contract.

1. Merge the two hooks into one. use chose to use a new hook, make sure it includes canBridge and allowance check